sdk/core/abort-controller/src/AbortSignal.ts

```diff
@@ -134,10 +134,23 @@ export class AbortSignal implements AbortSignalLike {
     * Dispatches a synthetic event to the AbortSignal.
     */
   dispatchEvent(event: Event): boolean {
-    const listeners = listenersMap.get(this)!;
-    listeners.forEach((listener) => {
-      listener.call(this, event);
-    });
+    if (event.type === 'abort') {
+      if (this.aborted) {
```

Comment:
This check shouldn't be done I think. `dispatchEvent` literally just dispatches - you can call dispatchEvent with an abort event, all your listeners will fire, and your signal will still not be aborted. Likewise an aborted signal can have "abort" dispatched on it, resulting in handlers being called again.

![image](https://user-images.githubusercontent.com/11236/63052083-5bbb9200-be93-11e9-8b62-6c2718460453.png)